### PR TITLE
[WNMGDS-2259] External link component

### PR DESCRIPTION
**Pull request by zarahzachz**
## Summary

WNMGDS-2259

[Demo url](https://cmsgov.github.io/design-system/branch/WNMGDS-2259/external-link-component)

Promote the third party external link pattern to a component. Update our docs to reflect this change and add references to Link and Button for increased awareness of this component.

### CODE REVIEW

File: packages/design-system/src/components/ThirdPartyExternalLink/ThirdPartyExternalLink.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+          actions={[
+            <Button variation="solid" key="external-link__confirm" href={href}>
+              OK
+            </Button>,
+            <Button
+              variation="ghost"
+              onClick={() => setShowDialog(false)}
+              key="external-link__cancel"
+            >
+              Cancel
```
Comment (pwolfert) [inline packages/design-system/src/components/ThirdPartyExternalLink/ThirdPartyExternalLink.tsx:46]: We'll want to use the i18n module for all this content. We can go ask product teams for approved Spanish translations for this the modal

**Review by pwolfert — COMMENTED** [review]
Excellent tests and a well written component! We just need to make the content internationalizable

File: packages/ds-medicare-gov/src/components/ThirdPartyExternalLink/ThirdPartyExternalLink.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { ThirdPartyExternalLink } from '@cmsgov/design-system';
+
+ThirdPartyExternalLink.defaultProps = {
+  ...ThirdPartyExternalLink.defaultProps,
+  learnMoreUrl:
+    'https://www.cms.gov/About-CMS/Agency-Information/Aboutwebsite/PolicyforLinkingtoOutsideWebsites',
```
Comment (zarahzachz) [inline packages/ds-medicare-gov/src/components/ThirdPartyExternalLink/ThirdPartyExternalLink.tsx:7]: This is a placeholder until I hear from Melissa.

**Review by forrestbaer — COMMENTED** [review]
This looks good! I'm guessing the i18n and analytics work will come later? I see that analytics are mentioned but I don't see how it operates.

**Review by pwolfert — APPROVED** [review]
LGTM 👍

**Pull request merged**